cli/connect/plain_text.go

```diff
@@ -0,0 +1,151 @@
+	"github.com/c-bata/go-prompt"
+	"github.com/tarantool/cartridge-cli/cli/common"
+)
+
+const (
+	tagPushPrefixYAML = `%TAG !push!`
+	tagPushPrefixLua  = `-- Push`
+)
+
+type ReadFromConnFunc func(net.Conn) ([]byte, error)
+
+func plainTextEval(console *Console, funcBodyFmt string, args ...interface{}) (interface{}, error) {
+	var plainTextEvalFunc func(conn net.Conn, funcBody string) (interface{}, error)
+
+	switch {
+	case console.outputMode == ConsoleYAMLOutput:
+		plainTextEvalFunc = common.EvalTarantoolConn
+	case console.outputMode == ConsoleLuaOutput:
+		plainTextEvalFunc = common.EvalTarantoolConnLua
+	default:
+		return nil, fmt.Errorf("Unknown output mode: %s", console.outputMode)
+	}
+
+	return plainTextEvalFunc(console.conn, fmt.Sprintf(funcBodyFmt, args...))
+}
+
+func getPlainTextCompleter(console *Console) prompt.Completer {
+	getSuggestionsPlainText := func(console *Console, lastWord string) interface{} {
+		res, err := console.Eval(
+			getSuggestionsPlainTextFuncBodyFmt,
+			lastWord, len(lastWord),
+		)
+
+		if err != nil {
+			return nil
+		}
+
+		return res
+	}
+
+	completer := func(in prompt.Document) []prompt.Suggest {
+		return getSuggestions(console, in, getSuggestionsPlainText)
+	}
+
+	return completer
+}
+
+func plainTextExecute(console *Console, in string) string {
+	var readFromConnFunc ReadFromConnFunc
+	if err := common.WriteToConn(console.conn, in+"\n"); err != nil {
+		log.Debugf("Failed to write to instance socket: %s", err)
+		log.Fatalf("Connection was closed. Probably instance process isn't running anymore")
+	}
+
+	console.conn.SetReadDeadline(time.Now().Add(1 * time.Second))
+
+	newOutputMode := getNewOutputMode(in)
+
+	switch {
+	case newOutputMode != "":
+		readFromConnFunc = getReadFromConnSetMode(console, newOutputMode)
+	case console.outputMode == ConsoleYAMLOutput:
+		readFromConnFunc = common.ReadFromConnYAML
+	case console.outputMode == ConsoleLuaOutput:
+		readFromConnFunc = common.ReadFromConnLua
+	default:
+		log.Fatalf("Unknown output mode: %s", console.outputMode)
+	}
+
+	var data string
+
+	for {
+		dataPortionBytes, err := readFromConnFunc(console.conn)
+		if err != nil {
+			log.Debugf("Failed to read from instance socket: %s", err)
+			log.Fatalf("Connection was closed. Probably instance process isn't running anymore")
+		}
+
+		dataPortion := string(dataPortionBytes)
+		data += dataPortion
+
+		if !pushTagIsReceived(console, dataPortion) {
+			break
+		}
+	}
+
+	return data
+}
+
+func pushTagIsReceived(console *Console, dataPortion string) bool {
+	switch {
+	case console.outputMode == ConsoleYAMLOutput:
+		return strings.HasPrefix(dataPortion, tagPushPrefixYAML)
+	case console.outputMode == ConsoleLuaOutput:
+		return strings.HasPrefix(dataPortion, tagPushPrefixLua)
+	}
+
+	return false
+}
+
+func getReadFromConnSetMode(console *Console, newOutputMode ConsoleOutputMode) ReadFromConnFunc {
+	readFromConnFunc := func(conn net.Conn) ([]byte, error) {
+		setModeResBytes := make([]byte, 1024)
```

Comment:
Here you don't read byte by byte as here https://github.com/tarantool/cartridge-cli/pull/425. Is it ok?

Reply:
Yes, here I read a result of `\set output <mode>` command. If everything before was read correclty, it should be `true` value or error message.

Reply:
I will add a comment here, just because your Birthday tomorrow. I sure that my comment is a great gift.

Reply:
> I sure that my comment is a great gift

The best gift is a gift that is made by own hands.